content/docs/concepts/security/index.md

```diff
@@ -239,6 +271,41 @@ Istio provides two types of authentication:
   original client making the request as an end-user or device. Istio
   supports authentication with JSON Web Token (JWT) validation.
 
+### Mutual TLS authentication
+
+Service-to-service communication is tunneled through the client side [Envoy](https://envoyproxy.github.io/envoy/) and the server side Envoy.
+For a client to call a server, the process can be described as the following:
+
+1. The outbound traffic from a client is rerouted to its local sidecar Envoy.
+
+1. The client side Envoy starts a mutual TLS handshake with the server side Envoy.
+   During the handshake, it also does a [secure naming](/docs/concepts/security/#secure-naming) check to verify that
+   the service account presented in the server certificate is authorized to run the target service.
```

Comment:
To avoid relative pronouns like "it" which are ambiguous, I suggest:
During the handshake, the client side Envoy also does a [secure naming](/docs/concepts/security/#secure-naming) check to verify that
   the service account presented in the server certificate is authorized to run the target service.